sdk/storage/azure-storage-blob/src/main/java/com/azure/storage/blob/models/ParallelTransferOptions.java

```diff
@@ -60,35 +68,54 @@ public ParallelTransferOptions(Integer blockSize, Integer numBuffers, ProgressRe
      * any data is sent. Must be greater than 0. May be null to accept default behavior, which is the maximum value the
      * service accepts for uploading in a single requests and is represented by
      * {@link BlockBlobAsyncClient#MAX_UPLOAD_BLOB_BYTES}.
+     * @deprecated Use fluent interface to set properties instead.
      */
+    @Deprecated
     public ParallelTransferOptions(Integer blockSize, Integer numBuffers, ProgressReceiver progressReceiver,
         Integer maxSingleUploadSize) {
-        if (blockSize != null) {
-            StorageImplUtils.assertInBounds("blockSize", blockSize, 1, BlockBlobAsyncClient.MAX_STAGE_BLOCK_BYTES);
-        }
-        this.blockSize = blockSize;
-
-        if (numBuffers != null) {
-            StorageImplUtils.assertInBounds("numBuffers", numBuffers, 2, Integer.MAX_VALUE);
-        }
-        this.numBuffers = numBuffers;
-        this.progressReceiver = progressReceiver;
-
-        if (maxSingleUploadSize != null) {
-            StorageImplUtils.assertInBounds("maxSingleUploadSize", maxSingleUploadSize, 1,
-                BlockBlobAsyncClient.MAX_UPLOAD_BLOB_BYTES);
-        }
-        this.maxSingleUploadSize = maxSingleUploadSize;
+        this.setBlockSize(blockSize == null ? null : Long.valueOf(blockSize));
+        this.setNumBuffers(numBuffers);
+        this.setProgressReceiver(progressReceiver);
+        this.setMaxSingleUploadSize(maxSingleUploadSize == null ? null : Long.valueOf(maxSingleUploadSize));
     }
 
     /**
      * Gets the block size (chunk size) to transfer at a time.
      * @return The block size.
+     * @deprecated Use {@link #getBlockSizeLong()}.
      */
+    @Deprecated
     public Integer getBlockSize() {
+        return this.blockSize == null ? null : Math.toIntExact(this.blockSize);
+    }
+
+    /**
+     * Gets the block size (chunk size) to transfer at a time.
+     * @return The block size.
+     */
+    public Long getBlockSizeLong() {
         return this.blockSize;
     }
 
+    /**
+     * Sets the block size (chunk size) to transfer at a time.
+     * For upload, The block size is the size of each block that will be staged. This value also determines the number
+     * of requests that need to be made. If block size is large, upload will make fewer network calls, but each
+     * individual call will send more data and will therefore take longer. This parameter also determines the size
+     * that each buffer uses when buffering is required and consequently amount of memory consumed by such methods may
+     * be up to blockSize * numBuffers.
+     *
+     * @param blockSize The block size.
+     * @return The ParallelTransferOptions object itself.
+     */
+    public ParallelTransferOptions setBlockSize(Long blockSize) {
```

Comment:
In such case I'd rather add non-null validation - but since it's optional customer should be able to "unset" it - end of day it just became mutable data structure. If I were customer where this thing is result of some computation (weird but possible) then I'd be quite upset about the need of if/else-ing. Besides if some of them are actually passing Integer around that could be null for some reason then autoboxing will blow up on nulls if they don't else/if correctly.

Other thing. Usually in classes like that I used to see symmetry between getter and setter - they usually speak same type. It'd be quite non-idiomatic in Java if we do otherwise.